Source/cursor.cpp

```diff
@@ -133,19 +142,34 @@ void FreeCursor()
 	ClearCursor();
 }
 
+const CelSprite &GetInvItemSprite(int i)
+{
+	return i < InvItems1Size ? *pCursCels : *pCursCels2;
+}
+
+int GetInvItemFrame(int i)
+{
+	return i < InvItems1Size ? i : i - (InvItems1Size - 1);
+}
+
+std::pair<int, int> GetInvItemSize(int i)
+{
+	if (i >= InvItems1Size)
+		return { InvItemWidth2[i - (InvItems1Size - 1)], InvItemHeight2[i - (InvItems1Size - 1)] };
+	return { InvItemWidth1[i], InvItemHeight1[i] };
+}
+
 void SetICursor(int i)
 {
-	icursW = InvItemWidth[i];
-	icursH = InvItemHeight[i];
+	std::tie(icursW, icursH) = GetInvItemSize(i);
 	icursW28 = icursW / 28;
 	icursH28 = icursH / 28;
 }
 
 void NewCursor(int i)
 {
 	pcurs = i;
-	cursW = InvItemWidth[i];
-	cursH = InvItemHeight[i];
+	std::tie(cursW, cursH) = GetInvItemSize(i);

```

Comment:
We should probably consider making some sort of structure for the cursor, there is also cursx/y afaik.